src/executor/thread_factory.cpp

```diff
@@ -40,7 +40,7 @@ class LightThread final : public IThread {
   }
 
   ~LightThread() final {
-    assert(_thread.get_id() != yaclib_std::this_thread::get_id());
+    YACLIB_ERROR(_thread.get_id() == yaclib_std::this_thread::get_id(), "join itself called");
```

Comment:
```suggestion
    YACLIB_ERROR(_thread.get_id() == yaclib_std::this_thread::get_id(), "Thread try to join itself, probably because you forgot Stop ThreadPool, and ThreadPool dtor was called in ThreadPool thread");
```